plugins/modules/system/kernel_blacklist.py

```diff
@@ -88,18 +89,21 @@ def state_absent(self):
             return
         self.vars.is_blacklisted = False
         self.vars.lines = [line for line in self.vars.lines if not self.pattern.match(line.strip())]
-        if self.has_changed() and not self.module.check_module:
-            with open(self.vars.filename, 'w') as fd:
-                fd.writelines(self.vars.lines)
 
     def state_present(self):
         if self.vars.is_blacklisted:
             return
         self.vars.is_blacklisted = True
         self.vars.lines = self.vars.lines + ['blacklist %s\n' % self.vars.name]
+
+    def __quit_module__(self):
         if self.has_changed() and not self.module.check_module:
-            with open(self.vars.filename, 'a') as fd:
-                fd.writelines(self.vars.lines[-1:])
+            dummy, tmpfile = tempfile.mkstemp()
```

Comment:
I think it's best to add the next lines in a try/finally block to remove tmpfile afterwards, in case it was left because the below code raises an exception. (Or simply add the filename to AnsibleModule's cleanup list.)

Reply:
Good call. Coming out in a couple of hours, along with some tests ;-)

Reply:
A couple of eight ;-) done now!